packages/aws-cdk/CONTRIBUTING.md

```diff
@@ -187,3 +187,32 @@ These two sets of integration tests have 3 running modes:
 The integration test and canary modes are used in the CDK publishing pipeline
 and the CDK canaries, respectively. You wouldn't normally need to run
 them directly that way.
+
+## Bundling
+
+Our CLI package is built and packaged using the [node-bundle](../../tools/%40aws-cdk/node-bundle/README.md) tool.
+
+This has two affects one should be aware of:
+
+### Runtime Dependencies
+
+All runtime dependencies are converted to `devDependencies`, as they are bundled inside
+the package and don't require installation by consumers. This process happens on-demand during packaging,
+this is why our [source code](./package.json) still contains those dependencies,
+but the [npm](https://www.npmjs.com/package/aws-cdk) package does not.
+
+### Attributions
+
+The bundler also creates an attributions document that lists out license information for the entire
+dependency closure. This document is stored in the [THIRD_PARTY_LICENSES](./THIRD_PARTY_LICENSES) file.
+Our build process validates that the file committed to source matches the expected auto-generated one.
+We do this so that our source code always contains the up to date attributions document, and so that we can
+backtrack/review changes to it using normal code review processes.
+
+Whenever a dependency changes (be it direct or transitive, new package or new version), the attributions document
+will change, and needs to be regenerated. For you, this means that:
+
+1. When you manually upgrade a dependency, you must also regenerate the document by running `yarn pkglint` inside the CLI package.
+2. When you build the CLI locally, you must ensure your dependencies are up to date by running `yarn install`.
```

Comment:
To avoid confusion we should probably mention whether this means running `yarn install` in the CLI package directory, or just at the root level.